### PR TITLE
feat: move the tenant system namespace inside the tenant

### LINKED ISSUE #3: Feature: add namespace to tenant namespaces
## Background
With the actual setup, the initial namespace created for the tenant is not part of the tenant
```
---
apiVersion: v1
kind: Namespace
metadata:
  name: oil-system
```

Causing the tenant owners not being able to list this namespace and manage resources within that namespace. In the flux example, tenant owners wont be able to handle the kustomization/gitrepository which is the root of their workloads.

## Proposal
Make the flux addon capable of modifying the owners reference of the namespace based in a label/annotation, ie: `capsule.clastix.io/tenant: oil`. Making this namespace part of the tenant so tenant owners can work with it.

### PR DESCRIPTION
**Pull request by maxgio92**
Fixes #3

### CODE REVIEW

File: e2e/serviceaccount_test.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ var _ = Describe("Creating a new ServiceAccount", func() {
 			})
 		})
 		Context("set as owner of the Tenant", func() {
-			BeforeEach(func() {
+			BeforeAll(func() {
```
Comment (maxgio92) [inline e2e/serviceaccount_test.go:93]: Now that the Tenant system NS is part of the Tenant, every time the Tenant is deleted also the system NS is going to be. Repeting creation and deletion of Tenant for every spec is not good and also, not needed.

File: e2e/serviceaccount_test.go [hunk 2]
```diff
@@ -255,6 +255,16 @@ var _ = Describe("Creating a new ServiceAccount", func() {
...
 					}, 20*time.Second, 1*time.Second).Should(Succeed())
 				})
+
+				It("should set the correct owner reference to the Tenant system Namespace", func() {
+					Eventually(func(g Gomega) {
+						ns := new(corev1.Namespace)
+						Expect(adminClient.Get(context.TODO(), types.NamespacedName{
+							Name: sa.Namespace,
+						}, ns)).Should(Succeed())
+						Expect(len(ns.OwnerReferences)).To(Equal(1))
```
Comment (maxgio92) [inline e2e/serviceaccount_test.go:265]: It needs further validation of the reference.

**Comment by maxgio92** [comment]
Need to lint 🫠

**Comment by maxgio92** [comment]
Wait for #7 to be merged and need to rebase on new mainline.

**Comment by maxgio92** [comment]
@prometherion the PR Is now rebased and readybfor review 🚀

File: pkg/controller/serviceaccount/namespaces.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+	capsulev1beta2 "github.com/projectcapsule/capsule/api/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Set the Tenant owner reference on the Namespace specified.
+func (r *ServiceAccountReconciler) setNamespaceOwnerRef(ctx context.Context, ns *corev1.Namespace, tnt *capsulev1beta2.Tenant) error {
+	scheme := runtime.NewScheme()
```
Comment (prometherion) [inline pkg/controller/serviceaccount/namespaces.go:17]: You can extract the Scheme from the controller runtime client already injected, if these ones are already registered in the manager setup.

**Review by prometherion — CHANGES_REQUESTED** [review]

File: pkg/controller/serviceaccount/namespaces.go (hunk 3 above)
Comment (maxgio92) [reply pkg/controller/serviceaccount/namespaces.go:17]: Thank you! TIL

Comment (maxgio92) [reply pkg/controller/serviceaccount/namespaces.go]: Applied @prometherion PTAL

**Pull request merged**